### PR TITLE
fix: fixes round state traced data to mirror the new state  

### LINKED ISSUE #1258: Fixing/verifying consensus round state traced data
Based on the findings reported in issue #1257, discrepancies have been noted in the recorded heights for `RoundStepPropose` and `RoundStepPrevote`. This issue aims to monitor the problem and determine whether the tracing data requires updates.

### PR DESCRIPTION
**Pull request by staheri14**
Closes #1258

### CODE REVIEW

**Comment by staheri14** [comment]
Blocking on #1260

**Review by rootulp — APPROVED** [review]
🤷‍♂️

File: consensus/state.go [hunk 1]
```diff
@@ -706,7 +706,9 @@ func (cs *State) newStep() {
 
 	cs.nSteps++
 
-	schema.WriteRoundState(cs.traceClient, cs.Height, cs.Round, cs.Step)
+	schema.WriteRoundState(cs.traceClient, cs.RoundState.Height,
+		cs.RoundState.Round,
+		cs.RoundState.Step)
```
Comment (evan-forbes) [inline consensus/state.go:711]: is this just being more explicit to where the height and round are coming from?

**Review by evan-forbes — APPROVED** [review]

File: consensus/state.go (hunk 1 above)
Comment (evan-forbes) [reply consensus/state.go:711]: not sure I would classify this as a fix as I currently understand this

**Comment by staheri14** [comment]
Closing this as the issue was already resolved.

**Pull request closed without merging**